test/integration/test_http.py

```diff
@@ -88,130 +102,115 @@ def test_basicTxn(self):
+        self.assertEqual(resultRec.fields.partitionKey, b"test1pk")
+        self.assertEqual(resultRec.fields.rangeKey, b"test1rk")
+        self.assertEqual(resultRec.fields.data, b"mydata")
 
         # Commit
-        request = {"txnID": txnId, "commit": True}
-        url = args.http + "/api/EndTxn"
-        r = requests.post(url, data=json.dumps(request))
-        result = r.json()
-        print(result)
-        self.assertEqual(result["status"]["code"], 200);
+        status = txn.end()
+        self.assertTrue(status.is2xxOK())
 
 
     def test_validation(self):
-        db = SKVClient(args.http)
-        # Create a location object
-        loc = DBLoc(partition_key_name="partitionKey", range_key_name="rangeKey",
-            partition_key="ptest2", range_key="rtest2",
-            schema="test_schema", coll="HTTPClient", schema_version=1)
-        additional_data =  { "data" :"data1"}
-
+        record = TestHTTP.schema.make_record(partitionKey=b"test2pk", rangeKey=b"test1rk", data=b"mydata")
         # Get a txn
-        status, txn = db.begin_txn()
-        self.assertEqual(status.code, 201)
+        status, txn = TestHTTP.cl.begin_txn()
+        self.assertTrue(status.is2xxOK())
 
         # Write/read with bad collection name, should fail
-        bad_loc = loc.get_new(coll="HTTPClient1")
-        status = txn.write(bad_loc, additional_data)
+        status = txn.write(b"HTTPClient1", record)
         self.assertEqual(status.code, 404)
-        status, record = txn.read(bad_loc)
+        status, _ = txn.read(b"HTTPClient1", record)
         self.assertEqual(status.code, 404)
 
-        # Write/Read with bad schemaName, should fail
-        bad_loc = loc.get_new(schema="test_schema1")
-        status = txn.write(bad_loc, additional_data)
+        # # Write/Read with bad schemaName, should fail
+        bad_loc = copy(record)
+        bad_loc.schemaName = b"test_schema1"
+        status = txn.write(TestHTTP.cname, bad_loc)
         self.assertEqual(status.code, 404)
-        status, record = txn.read(bad_loc)
+        status, _ = txn.read(TestHTTP.cname, bad_loc)
         self.assertEqual(status.code, 404)
 
         # Write with bad schema version, should fail
-        bad_loc = loc.get_new(schema_version=2)
-        status = txn.write(bad_loc, additional_data)
+        bad_loc = copy(record)
+        bad_loc.schemaVersion = 2
+        status = txn.write(TestHTTP.cname, bad_loc)
         self.assertEqual(status.code, 404)
 
         # Write/Read with bad partition key data type, should fail
-        bad_loc = loc.get_new(partition_key=1)
-        status = txn.write(bad_loc, additional_data)
-        self.assertEqual(status.code, 400)
-        status, _ = txn.read(bad_loc)
-        self.assertEqual(status.code, 400)
+        bad_loc = TestHTTP.schema.make_record(partitionKey=1, rangeKey=b"test1rk", data=b"mydata")
+        status = txn.write(TestHTTP.cname, bad_loc)
+        self.assertEqual(status.code, 503)
```

Comment:
this is a problem i think - why is the code 503?

Reply:
Fixed